nanoFramework.System.Net.Http/Http/HttpContent.cs

```diff
@@ -0,0 +1,261 @@
+        /// <remarks>
+        /// <para>
+        /// This operation will block.
+        /// </para>
+        /// <para>
+        /// This is the .NET nanoFramework equivalent of ReadAsByteArrayAsync.
+        /// </para>
+        /// </remarks>
+        public byte[] ReadAsByteArray()
+        {
+            LoadIntoBuffer();
+
+            return _buffer.ToArray();
+        }
+
+        /// <summary>
+        /// Serialize the HTTP content to a string as an synchronous operation.
+        /// </summary>
+        /// <returns>A string with the HTTP content.</returns>
+        /// <remarks>
+        /// <para>
+        /// This operation will block.
+        /// </para>
+        /// <para>
+        /// This supports only <see cref="Encoding.UTF8"/> encoding.
+        /// </para>
+        /// <para>
+        /// This is the .NET nanoFramework equivalent of ReadAsStringAsync.
+        /// </para>
+        /// </remarks>
+        public string ReadAsString()
+        {
+            LoadIntoBuffer();
+
+            if (_buffer.Length == 0)
+            {
+                return string.Empty;
+            }
+
+            // we only support UTF-8
+            return Encoding.UTF8.GetString(
+                _buffer.GetBuffer(),
+                0,
+                (int)_buffer.Length);
+        }
+
+        /// <summary>
+        /// Determines whether the HTTP content has a valid length in bytes.
+        /// </summary>
+        /// <param name="length">The length in bytes of the HTTP content.</param>
+        /// <returns><see langword="true"/> if <paramref name="length"/> is a valid length; otherwise, <see langword="false"/>.</returns>
+        protected internal abstract bool TryComputeLength(out long length);
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            Dispose(true);
+        }
+
+        /// <inheritdoc/>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing && !_disposed)
+            {
+                _disposed = true;
+
+                if (_buffer != null)
+                {
+                    _buffer.Dispose();
+                }
+            }
+        }
+
+        /// <summary>
+        /// When overridden in a derived class, serializes the HTTP content to a stream. Otherwise, throws a <see cref="NotSupportedException"/>.
+        /// </summary>
+        /// <param name="stream">The target stream.</param>
+        /// <returns></returns>
+        /// <exception cref="NotSupportedException">The method is not overridden in the derived class.</exception>
+        /// <remarks>
+        /// <para>
+        /// This is the .NET nanoFramework equivalent of SerializeToStream(Stream, TransportContext, CancellationToken).
+        /// </para>
+        /// </remarks>
+        protected abstract void SerializeToStream(Stream stream);
+
+        sealed class FixedMemoryStream : MemoryStream
+        {
+            readonly long _maxSize;
```

Comment:
most likely private

Reply:
you are still missing the explicit visibility

Reply:
I'm seeing it as private.... 😯